client/src/Tickets/TicketModal.jsx

```diff
@@ -84,26 +86,78 @@ function Charger({ stateChange }) {
   );
 }
 
-const AddTicketModel = ({ update, data, stateChange, toggleUserModal, setToggleUserModal }) => {
+const TicketModal = ({ onCreate, onUpdate }) => {
+  const navigate = useNavigate();
+  const { ticketId } = useParams();
+  const [data, setData] = useState({
+    ClientId: null,
+    LocationId: null,
+    UserId: null,
+    serialNumber: '',
+    device: '',
+    problem: '',
+    troubleshooting: '',
+    resolution: '',
+    dateOn: DateTime.now().toISODate(),
+    timeInAt: '',
+    timeOutAt: '',
+    hasCharger: false,
+    notes: '',
+  });
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await fetch(`/api/tickets/${ticketId}`);
+      if (response.ok) {
+        const data = await response.json();
+        setData(data);
+      }
+    }
+    if (ticketId) {
+      fetchData();
+    }
+  }, [ticketId]);
```

Comment:
If we're editing an existing ticket, fetch it from the API...

Reply:
It may seem redundant to re-fetch the data from the API, but in passing it from the list/table to the modal, you're assuming that you'd always be loading all the data you need for each record in the list/table...

As a system becomes more complex and each record has more data, you might end up optimizing the list/table fetch so that it only returns the data that you need to render the list/table (i.e. only what's visible in the columns)- so you wouldn't have all the data to display/edit in the modal.